### PR TITLE
bibtool: added year to citekey

### PR DESCRIPTION
**Pull request by marcfehling**
Part of #389. This should fix part 2 of #405.

I've changed the citekey that in starts with the year.

The nice thing about this approach is that the publications are now also sorted by year first, which lets us easily identify articles that are in the wrong file. I moved some articles around in the second commit and found a few duplicates.

### CODE REVIEW

File: publications-2015.bib [hunk 1]
```diff
@@ -567,6 +567,17 @@ @Article{2015:grieshaber.mcbride.ea:uniformly
   pages   = {2255--2278}
 }
 
+@InProceedings{2015:hai.bause:adaptive,
+  doi     = {10.1115/imece2015-53265},
+  url     = {https://doi.org/10.1115/imece2015-53265},
+  year    = 2015,
+  month   = nov,
```
Comment (marcfehling) [inline publications-2015.bib:574]: The conference happened in Nov 15, but the article was first published online Mar 2016.

File: publications-2018.bib [hunk 2]
```diff
@@ -1,74 +1,5 @@
...
-  eprint  = {1708.01773},
-  archiveprefix = {arXiv},
-  primaryclass = {cs.CE}
-}
-
-@Article{2017:emerson.farrell.ea:computing,
-  doi     = {10.1080/02678292.2017.1365385},
-  url     = {https://doi.org/10.1080/02678292.2017.1365385},
-  year    = 2017,
-  month   = aug,
```
Comment (marcfehling) [inline publications-2018.bib]: Here and for the following: The journals have been published in 2018, but they have been online already in 2017.

File: publications-2015.bib (hunk 1 above)
Comment (bangerth) [reply publications-2015.bib:574]: As a general rule, please do one thing at a time. It's impossible to look through thousands of lines of changes that are automatically generated if you're also making semantic changes. If you need to make a change like this, do it in a separate patch.

Comment (marcfehling) [reply publications-2015.bib:574]: It's in the second commit. I can easily turn it into a separate PR if you like.

**Review by bangerth — APPROVED** [review]

**Pull request merged**